src/order/bounded_lattice.lean

```diff
@@ -2,16 +2,36 @@
 Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl
-
-Defines bounded lattice type class hierarchy.
-
-Includes the Prop and fun instances.
 -/
 import order.lattice
 import data.option.basic
 import tactic.pi_instances
 import logic.nontrivial
 
+/-!
+# ⊤ and ⊥, bounded lattices and variants
+
+This file defines top and bottom elements (greatest and least elements) of a type, the bounded
+variants of different kinds of lattices, sets up the typeclass hierarchy between them and provides
+instances for `Prop` and `fun`.
+
+## Main declarations
+
+* `has_<top/bot> α`: Typeclasses to declare the `⊤`/`⊥` notation.
+* `order_<top/bot> α`: Order with a top/bottom element.
+* `with_<top/bot> α`: Equips `option α` with the order on `α` plus `none` as the top/bottom element.
+* `semilattice_<sup/inf>_<top/bot>`: Semilattice with a join/meet and a top/bottom element. Typical
+  examples include `ℕ`.
```

Comment:
Maybe make clearer here that the `sup/inf` and `top/bot` are independent choices, which is not immediately obvious?